modules/javafx.controls/src/main/java/javafx/scene/control/ControlUtils.java

```diff
@@ -105,6 +106,7 @@ static void requestFocusOnControlOnlyIfCurrentFocusOwnerIsChild(Control c) {
             }
 
             @Override public int getRemovedSize() {
+                checkState();
```

Comment:
Is this needed here? Just wondering because its was not there before.

Reply:
Without the check, it would be possible to inspect the size without calling `next()` first.

Reply:
Right. But I want to understand why this was not there before. 

Is this needed, because if you didn't called `next()`, there is no change yet to retrieve the removed size?

Reply:
It is needed according to the documentation of `ListChangeListener.Change`:

_Important: It's necessary to call next() method before calling any other method of Change. The same applies after calling reset(). The only methods that works at any time is getList()._